RELEASE.md

```diff
@@ -0,0 +1,69 @@
+# Node.js Releases
+
+<!-- TODO(joyeecheung): merge it into Release#readme? -->
```

Comment:
I'm not sure this belongs to the root level – but yeah indeed it's worth considering whether it makes sense to keep it in this repo. FWIW I've suggested putting it on the website in https://github.com/nodejs/nodejs.org/issues/7942.